cpp/test/compat/comm/commcompat.cpp

```diff
@@ -22,7 +24,9 @@ using namespace unittest::compat;
 FILE* popen_wrapper(const char* command, const char* mode)
 {
 #ifdef _WIN32
-    return _popen(command, mode);
+    std::string escaped_command = command;
+    boost::replace_all(escaped_command, " ", "\" \"");
```

Comment:
Am I reading this correctly: replace a single pace with the string `" "` (quote space quote)? That doesn't seem quite right. Say we want to execute the program `cool beans.exe` and pass it the file `how now.txt`.

    popen_wrapper("cool beans.exe how now.txt", "r");

I expect:

    "cool beans.exe" "how now.txt"

But this code looks like it would produce:

    cool" "beans.exe" "how" "now.txt

I think the quoting needs to happen one level above, in CMake, where it knows what the individual parts of the arguments are. By the time we're here, I think we've lost the distinction between what is the command and what is an argument.